sklearn/impute.py

```diff
@@ -39,8 +40,9 @@
 
 def _get_mask(X, value_to_mask):
     """Compute the boolean mask X == missing_values."""
-    if value_to_mask == "NaN" or np.isnan(value_to_mask):
-        return np.isnan(X)
+    if value_to_mask is np.nan:
+        # nan values are never equal to themselves
```

Comment:
I would add that this way it also works for object dtypes (in case somebody would wonder why not np.isnan is used)